### PR TITLE
[nGraph][IE] Enable FP64 data type in IE and review nGraph Python tests

### PR DESCRIPTION
**Pull request by jiwaszki**

### CODE REVIEW

File: inference-engine/include/ie_precision.hpp [hunk 1]
```diff
@@ -18,411 +18,419 @@
...
+    BIN = 71,  /**< 1bit integer value */
+    BOOL = 41, /**< 8bit bool type */
+    CUSTOM = 80 /**< custom precision has it's own name and size of elements */
+  };
 
 private:
-    struct PrecisionInfo {
-        /** @brief Size of underlined element */
-        size_t bitsSize = 0;
+  struct PrecisionInfo {
```
Comment (ilyachur) [inline inference-engine/include/ie_precision.hpp:52]: Why did you change indentation?

File: inference-engine/include/ie_precision.hpp [hunk 2]
```diff
@@ -18,411 +18,419 @@
...
-        U32 = 74,          /**< 32bit unsigned integer value */
-        I64 = 72,          /**< 64bit signed integer value */
-        U64 = 73,          /**< 64bit unsigned integer value */
-        BIN = 71,          /**< 1bit integer value */
-        BOOL = 41,         /**< 8bit bool type */
-        CUSTOM = 80        /**< custom precision has it's own name and size of elements */
-    };
+  /** Enum to specify of different  */
+  enum ePrecision : uint8_t {
+    UNSPECIFIED = 255, /**< Unspecified value. Used by default */
```
Comment (ilyachur) [inline inference-engine/include/ie_precision.hpp:28]: Why did you change the style?

**Review by ilyachur — COMMENTED** [review]
It is hard to review because you changed the code style for all file.

In general if you add a new precision to the Inference Engine please also add new tests.
You can use this PR as an example: https://github.com/openvinotoolkit/openvino/pull/1297/files

**Review by tsocha — CHANGES_REQUESTED** [review]
please revert indentation changes to keep diff tidy.

File: inference-engine/include/ie_precision.hpp (hunk 1 above)
Comment (jiwaszki) [reply inference-engine/include/ie_precision.hpp:52]: I used default code formatter that probably messed up the file. I'm reverting it and adding tests.
@tsocha

**Review by postrational — APPROVED** [review]
Please revert indentation changes. Otherwise looks good.

**Comment by mbencer** [comment]
Should be `fp64` type added to `ie_api.pyx` (https://github.com/openvinotoolkit/openvino/blob/0c79804bcd96314fefc0ea54663c04b9e8fddcd6/inference-engine/ie_bridges/python/src/openvino/inference_engine/ie_api.pyx#L1653) is we start support it?
@akuporos What do you think?

File: inference-engine/ie_bridges/python/src/openvino/inference_engine/constants.pyx [hunk 3]
```diff
@@ -18,7 +18,7 @@ from .cimport ie_api_impl_defs as C
 import numpy as np
 from enum import Enum
 
-supported_precisions = ["FP32", "FP16", "I64", "U64", "I32", "U32", "I16", "I8", "U16", "U8"]
+supported_precisions = ["FP32", "FP64", "FP16", "I64", "U64", "I32", "U32", "I16", "I8", "U16", "U8"] # ???
```
Comment (ilyachur) [inline inference-engine/ie_bridges/python/src/openvino/inference_engine/constants.pyx:21]: I think we shouldn't merge a `# ???` comment to master.

I think it will be better to ask @akuporos about this code.

File: inference-engine/src/legacy_api/src/ie_cnn_layer_builder_ngraph.cpp [hunk 4]
```diff
@@ -382,6 +382,9 @@ CNNLayer::Ptr NodeConverter<ngraph::op::Convert>::createLayer(const std::shared_
     case Precision::FP32:
         precision_str = "FP32";
         break;
+    case Precision::FP64:
```
Comment (ilyachur) [inline inference-engine/src/legacy_api/src/ie_cnn_layer_builder_ngraph.cpp:385]: It can be done separately.

I think we can replace this switch/case to one line: `precision_str = p.name();`

**Review by tomdol — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Comment by ilyachur** [comment]
Please fix a CI.

**Review by mbencer — APPROVED** [review]

**Review by akuporos — APPROVED** [review]

**Review by akuporos — COMMENTED** [review]
It'd be nice to add a test [here](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/ie_bridges/python/tests/test_Blob.py)

**Comment by ilyachur** [comment]
Please make CI green.

**Review by tsocha — APPROVED** [review]
